### PR TITLE
remove `zip(strict=True)` in k-nearest p-median

### LINKED ISSUE #402: k-nearest pmedian failures due to `zip(strict=True)`
[2023-10-13 CI – py38](https://github.com/pysal/spopt/actions/runs/6514516150/job/17695823053#step:6:630)

k-nearest pmedian failures due to `zip(strict=True)`

```
FAILED spopt/tests/test_knearest_p_median.py::TestKNearestPMedian::test_knearest_p_median_from_geodataframe - TypeError: zip() takes no keyword arguments
FAILED spopt/tests/test_knearest_p_median.py::TestKNearestPMedian::test_knearest_p_median_from_geodataframe_no_results - TypeError: zip() takes no keyword arguments
FAILED spopt/tests/test_knearest_p_median.py::TestKNearestPMedian::test_solve - TypeError: zip() takes no keyword arguments
```

This seems to have flown under the radar due to #386. I will remove the `strict=True` keyword for now. We can re-add once `python=3.{8,9}` are dropped and python=3.12 is added , which will be soon. I anticipate this for a `v0.6.1` release.

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR resolves #402 and runs `pmedian.ipynb` for output.

### CODE REVIEW

**Comment by jGaboardi** [comment]
Can confirm that all tests are passing/failing as they should (considering #386).

File: spopt/locate/p_median.py [hunk 1]
```diff
@@ -806,8 +806,10 @@ def _from_sparse_matrix(self) -> None:
 
         # Client assignment integer decision variables
         row_indices, col_indices, values = find(self.aij)
+
+        # TODO: add `strict=True` back in to `zip` – gh-spopt#402
```
Comment (martinfleis) [inline spopt/locate/p_median.py:810]: I'd remove this. We're zipping an output of scipy.sparse.find. That is never of different length so using strict here has no reason.

Comment (jGaboardi) [reply spopt/locate/p_median.py:810]: Good point. I'll remove the note and add a `# noqa`

Comment (martinfleis) [reply spopt/locate/p_median.py]: Why `noqa`?

Comment (jGaboardi) [reply spopt/locate/p_median.py]: > Why `noqa`?

[`B905`](https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/) is raised without it.

Comment (jGaboardi) [reply spopt/locate/p_median.py]: @martinfleis Is there something I am missing that makes this a bad idea?

Comment (martinfleis) [reply spopt/locate/p_median.py]: > [B905](https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/) is raised without it.

Is it? I don't see it locally and it should not, given we specify `target-version = "py38"`. Ruff should not suggest to include a keyword that does not exist in 3.8.

Comment (martinfleis) [reply spopt/locate/p_median.py]: It doesn't matter in practice, it is just strange that you see this. I don't mind to have it there, I just think that right now it should not raise and once we update min Python, Ruff will tell us to add a keyword here. But not now.

Comment (jGaboardi) [reply spopt/locate/p_median.py]: Fair enough. I am being proactive with it. No need to do so, I suppose. I'll remove and merge.

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**